### PR TITLE
Dockerfile for cross compilation

### PR DESCRIPTION
**Pull request by crazy-max**
Follow-up #209 to split a bit what was done.

This one adds a Dockerfile to ease cross compilation and a simple GHA workflow that will just build the Dockerfile. Jenkinsfile has been removed as it's not used anymore. In another follow-up we can move the travis pipeline to GHA.

Build example: https://github.com/crazy-max/docker-credential-helpers/actions/runs/2288625907

cc @thaJeztah @ebriney

### CODE REVIEW

**Review by ebriney — APPROVED** [review]
Nice! LGTM

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  xx-go build -ldflags "$(cat /tmp/.ldflags)" -o /out/docker-credential-wincred-${TARGETARCH}${TARGETVARIANT}.exe ./wincred/cmd/main_windows.go
+  xx-verify /out/docker-credential-wincred-${TARGETARCH}${TARGETVARIANT}.exe
+EOT
+
+FROM build-$TARGETOS AS build
+
+FROM scratch AS binaries
+COPY --from=build /out /
+
+FROM binaries
```
Comment (thaJeztah) [inline Dockerfile:82]: 👀 why the `FROM` here? 🤔

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,82 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_VERSION=1.16.7
```
Comment (thaJeztah) [inline Dockerfile:3]: (After this?) we should update to a current version of Go

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ARG XX_VERSION=1.1.2
+ARG OSXCROSS_VERSION=11.3-r7-alpine
+
+ARG PKG=github.com/docker/docker-credential-helpers
+
+# xx is a helper for cross-compilation
+FROM --platform=$BUILDPLATFORM tonistiigi/xx:${XX_VERSION} AS xx
+
+# osxcross contains the MacOSX cross toolchain for xx
+FROM crazymax/osxcross:${OSXCROSS_VERSION} AS osxcross
```
Comment (thaJeztah) [inline Dockerfile:13]: This is built from https://github.com/crazy-max/osxcross ? Or from the upstream one? (I see neither repo tags releases, so was wondering what version it was built from)

**Review by thaJeztah — COMMENTED** [review]
overall looks good (thanks!) some questions

File: Dockerfile (hunk 2 above)
Comment (crazy-max) [reply Dockerfile:3]: Yes just keeping minimal atm

File: Dockerfile (hunk 3 above)
Comment (crazy-max) [reply Dockerfile:13]: > This is built from https://github.com/crazy-max/osxcross ?

It's built from https://github.com/crazy-max/docker-osxcross

File: Dockerfile (hunk 1 above)
Comment (crazy-max) [reply Dockerfile:82]: To make it the default one when building. I could also just `FROM scratch` the previous one but I tend to prefer naming stages so the bake definition fits better.

Comment (crazy-max) [reply Dockerfile:82]: Oh nevermind I can just remove it I think

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**